Lib/idlelib/idle_test/test_squeezer.py

```diff
@@ -170,6 +170,7 @@ def test_write_not_stdout(self):
 
     def test_write_stdout(self):
         """Test Squeezer's overriding of the EditorWindow's write() method."""
+        requires('gui')
```

Comment:
It makes no sense to me that this should have any effect as this test should not require a gui and does not on all other CI and buildbots that do not have such.  Indeed, if Apple tkinter is being used, I expect `requires('gui')` should be True, and this should also have no effect.  This could be conditioned on `if macosx.isAquaTk():`, which is already used elsewhere in this file.

Reply:
Yeah, I was confused by this, too. But it was clear from the buildbot log that all the other squeezer tests were being properly skipped due by failing their requires('gui') check as expected. The one thing unique to `test_write_stdout` is that it has two calls to `make_squeezer_instance()` and it is the second call that doesn't properly fail, i.e. it proceeds to actually create a real Tk() root and that's what causes the segfault under these circumstances. As best I can tell, that is still happening on `main` and such but, with an up-to-date Tk 8.6.13(), that call doesn't cause a crash anymore. An edge case, in any case.